### PR TITLE
hir: Refactor getters for owner nodes

### PR DESCRIPTION
**Pull request by petrochenkov**

### CODE REVIEW

File: tests/incremental/hashes/inherent_impls.rs [hunk 1]
```diff
@@ -53,7 +53,7 @@ impl Foo {
 #[cfg(not(any(cfail1,cfail4)))]
 #[rustc_clean(cfg="cfail2")]
 #[rustc_clean(cfg="cfail3")]
-#[rustc_clean(cfg="cfail5")]
+#[rustc_clean(cfg="cfail5", except="opt_hir_owner_nodes")]
```
Comment (petrochenkov) [inline tests/incremental/hashes/inherent_impls.rs:56]: I'm not sure what made the incremental tests change.

Comment (petrochenkov) [reply tests/incremental/hashes/inherent_impls.rs:56]: cc @cjgillot

Comment (oli-obk) [reply tests/incremental/hashes/inherent_impls.rs:56]: Before this PR `MaybeOwner::Phantom` was returned either when that was the entry (can that happen?) or when there was no entry. Now there is a difference between `None` and `MaybeOwner::Phantom`.

Comment (petrochenkov) [reply tests/incremental/hashes/inherent_impls.rs:56]: Hmm, the current logic is `tcx.hir_crate(()).owners.get(id)?.as_owner().map(|i| &i.nodes)`, that means `None` is returned for all cases without actual `.nodes` (no entry, `Phantom` entry, `NonOwner` entry).

So, I'd expect the opposite effect - *more* clean queries, not less.

Comment (cjgillot) [reply tests/incremental/hashes/inherent_impls.rs:56]: This is due to relative spans, which are sensitive to the number of characters in the item's span. By adding the opt_ prefix, you have more characters in inside the curly braces than the original version has. This is the reason for the silly filler comments. 

TBH, I'm starting to reconsider the usefulness of these filler comments, and go towards testing the realistic version.

**Comment by petrochenkov** [comment]
Ok, I have updated the padding comments in incremental tests.
@rustbot ready

**Pull request merged**